static/shared/js/typeahead.js

```diff
@@ -131,11 +131,11 @@ export function triage(query, objs, get_item = (x) => x) {
 }
 
 export function sort_emojis(objs, query) {
-    const lowerQuery = query.toLowerCase();
+    query = query.toLowerCase();
```

Comment:
I think this change accidentally has functional changes -- the `triage` function prioritizes exact matches and case-sensitive prefix matches over case-insensitive matches, and lower-casing what we pass to it in advance will break that.

I think there's a valid argument that we intend for emoji in particular to be fully case-insensitive -- I guess the corner case would be if someone adds a "Capitalized" custom emoji and wants to find that immediately by typing "C".  But certainly the behavior of `triage` is not case-insensitive.

Reply:
Hmm, I guess that commit message is a bit too glib.

But yeah, I think to the extent this is changing any behavior, it's basically fixing a glitch.

If someone adds an emoji with a name like "GitHub", or "LaTeX", then people trying to find that are naturally going to type in lowercase, because that's how virtually every search UI in the computing universe works. (Even the option of case-sensitivity is basically unique to search UIs aimed at programmers searching through code; and even there it's usually off by default.)

Then the current behavior means that those matches get weirdly taken out of order and moved to the end. If there are more than 8 results, the effect is that they don't show up at all, while others with similar names do show up.

I'll revise the commit message.

Reply:
Yeah, that sounds right to me!